File: SmartApp/app/src/main/java/cn/cmy/smartapp/persenter/BasePresenter.java
package cn.cmy.smartapp.persenter;

import cn.cmy.smartapp.view.BaseView;

public abstract class BasePresenter<V extends BaseView> {

    private V v;

    public void attachView(V v) {
        this.v = v;
    }

    public void detachView() {
        //终止请求
        this.v = null;
    }

    public V getV(){
        return v;
    }
}

File: mvp/presenter/BasePresenter.kt
package com.cmy.mvp.presenter

import com.cmy.mvp.view.IBaseView

open abstract class BasePresenter<V : IBaseView> {

    var v:V? = null

    fun attach(v:V){
        this.v = v
    }

    fun detach(){
        this.v = null
    }

}